tests/AccountTests.test.ts

```diff
@@ -0,0 +1,248 @@
+        ).rejects.toThrow(
+            new LoernwerkError(
+                'Incorrect Password',
+                LoernwerkErrorCodes.NOT_FOUND
+            )
+        );
+    });
+
+    //getAccountById function
+    it('get existing account', async () => {
+        const mockUser = await DBUser.findBy({ name: 'magnus' });
+        expect(mockUser.length).toEqual(1);
+        const returnedUser = await AccountController.getAccountById(
+            mockUser[0].id
+        );
+        const hashComparison = await bcrypt.compare(
+            'verySecurePW',
+            returnedUser.password
+        );
+        expect(returnedUser).toBeInstanceOf(DBUser);
+        expect(returnedUser.name).toEqual('magnus');
+        expect(hashComparison).toEqual(true);
+        expect(returnedUser.mail).toEqual('magnus@carlsen.de');
+        expect(returnedUser.id).not.toEqual(null);
+    });
+
+    it('try getting nonexistent account', async () => {
+        await expect(() =>
+            AccountController.getAccountById(correctUser.id)
+        ).rejects.toThrow(
+            new LoernwerkError(
+                'No existing User with given ID',
+                LoernwerkErrorCodes.NOT_FOUND
+            )
+        );
+    });
+
+    //getAllAccounts function
+    it('get all accounts', async () => {
+        const allAccounts = await AccountController.getAllAccounts();
+        expect(allAccounts[0]).toBeInstanceOf(DBUser);
+        expect(allAccounts[0].name).not.toBe(null);
+        expect(allAccounts[0].id).not.toBe(null);
+    });
+
+    //deleteAccount function
+    it('delete a correct account', async () => {
+        const accountToBeDeleted = await DBUser.findBy({ name: 'bobby' });
+        await AccountController.deleteAccount(accountToBeDeleted[0].id);
+        await expect(() =>
+            SequenceController.getSequenceByCode('code66')
+        ).rejects.toThrow(
+            new LoernwerkError(
+                'No matching sequence',
+                LoernwerkErrorCodes.NOT_FOUND
+            )
+        );
+    });
+
+    /**
+     * ensureAdminExists function
+     *NOTE: Admin email returns undefined@loernwerk.de, intended?
+     */
+    it('create Admin, because no admin exists', async () => {
+        await AccountController.ensureAdminAccount();
+        const admin = await DBUser.findBy({ type: UserClass.ADMIN });
+        expect(admin[0]).toBeInstanceOf(DBUser);
+        expect(admin[0]).toHaveProperty('type', UserClass.ADMIN);
+    });
+
+    /**
+     * should undefined be returned if admin already exists?
+     * NOTE: This test only runs in suite, because admin was created in prior test
+     */
+    it('create Admin, admin already exists', async () => {
+        const existingAdmin = await DBUser.findBy({ type: UserClass.ADMIN });
+        expect(existingAdmin[0]).toHaveProperty('type', UserClass.ADMIN);
+        const admin = await AccountController.ensureAdminAccount();
```

Comment:
ensureAdminAccount always returns `undefined`, no matter if an admin was created or not, so I'm not sure if this is testable here